nova_vm/src/engine/bytecode/vm.rs

```diff
@@ -1264,6 +1251,34 @@ impl Vm {
                     result?;
                 }
             }
+            Instruction::IteratorStepValueOrUndefined => {
+                let iterator = vm.iterator_stack.last_mut().unwrap();
+                let result = iterator.step_value(agent);
+                if let Ok(result) = result {
+                    vm.result = Some(result.unwrap_or(Value::Undefined));
+                    if result.is_none() {
+                        // We have exhausted the iterator; replace it with the empty VmIterator so
+                        // further instructions aren't observable.
+                        *iterator = VmIterator::EmptyIterator;
+                    }
+                } else {
+                    vm.iterator_stack.pop();
+                    result?;
+                }
+            }
+            Instruction::IteratorRestIntoArray => {
+                let mut iterator = vm.iterator_stack.pop().unwrap();
+                let capacity = iterator.remaining_length_estimate(agent).unwrap_or(0);
+                let array = array_create(agent, 0, capacity, None)?;
+
+                let mut idx: u32 = 0;
+                while let Some(value) = iterator.step_value(agent)? {
+                    let key = PropertyKey::Integer(idx.into());
+                    create_data_property_or_throw(agent, array, key, value)?;
```

Comment:
**thought:** I'm pretty sure this cannot throw under any circumstances, right? We would maybe benefit from an `Array::push` method that could be used directly in these cases.

Reply:
You're right that this can't throw, I somehow ended up copying this line from the `ArrayPush` instruction and failed to notice that the spec explicitly marks this operation with `!`.